plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibraries.java

```diff
@@ -64,6 +87,63 @@ public static Library getLibrary(String id) {
     return libraries.get(id);
   }
 
+  private final static Supplier<List<Library>> clientApis = Suppliers.memoize(new Supplier<List<Library>>() {
+    public List<Library> get() {
+      Bundle bundle = FrameworkUtil.getBundle(CloudSdk.class);
+      URL url = bundle.getResource("/com/google/cloud/tools/libraries/libraries.json");
+      
+      try (InputStream in = url.openStream()) {
+        JsonReaderFactory factory = Json.createReaderFactory(null);
+        JsonReader reader = factory.createReader(in); 
+        JsonObject[] apis = reader.readArray().toArray(new JsonObject[0]); 
+        List<Library> clientApis = new ArrayList<>(apis.length);
+        for (JsonObject api : apis) {
+          String name = api.getString("name");
+          String id = api.getString("id");
+          Library library = new Library(id);
+          library.setGroup(CLIENT_APIS_GROUP);
+          library.setName(name);
+          JsonArray clients = api.getJsonArray("clients");
+          for (JsonObject client : clients.toArray(new JsonObject[0])) {
+            JsonString language = client.getJsonString("language");
+            if (language != null && "java".equals(language.getString())) {
+              String toolTip = client.getString("infotip");
+              library.setToolTip(toolTip);
+              JsonObject coordinates = client.getJsonObject("mavenCoordinates");
+              String groupId = coordinates.getString("groupId");
+              String artifactId = coordinates.getString("artifactId");
+              ArtifactVersion version =
+                  ArtifactRetriever.DEFAULT.getLatestArtifactVersion(groupId, artifactId);
```

Comment:
This always tries to use the latest version, which will almost all the time succeed, instead of the versions curated in `libraries.json`. Is this intended?

Reply:
Yes, it's intended. I'm assuming the metadata will go out of date.